plugins/modules/source_control/gitlab/gitlab_runner.py

```diff
@@ -57,6 +57,12 @@
       - The registration token is used to register new runners.
     required: True
     type: str
+  owned:
+    description:
+      - Searches only owned runners when searching for existing, when false admin token required.
```

Comment:
```suggestion
      - Searches only owned runners (i.e. specific and group runners) when searching for existing runners, when false admin token required, but also returns shared runners.
```

In one of the original issues about GitLab shared runners (https://github.com/ansible/ansible/issues/60635), @jklaiho writes:

> As far as I have been able to discover, the issue happens because of how `gitlab_runner.py:262` (`findRunner`) operates. It uses the `gitlab.GitLab.runners.list()` method, which only lists "owned runners" (quoted from https://python-gitlab.readthedocs.io/en/stable/gl_objects/runners.html#global-runners-admin). Shared runners are not listed. Listing them requires using `gitlab.GitLab.runners.all()`, which may require the access token of an admin-level user.

The latest [GitLab Runner docs](https://docs.gitlab.com/ee/ci/runners/README.html) describe shared, specific, and group runners, but make no mention of "owned" runners.

To the best of my understanding, the python-gitlab module is using the word "owned" to describe specific and group runners, whereas shared runners are separate from that category.

Feel free to change up the language if you think it can be made more clear.

Reply:
I'm going to switch this back to owned. It's what's used in the [API Docs](https://docs.gitlab.com/ee/api/runners.html#list-owned-runners)

Reply:
Sounds good. Thanks for looking closer.